### PR TITLE
Arweave Metadata

### PR DESCRIPTION
**Pull request by matt-malarkey**
@DJRHails another one for you 😉

### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [ ] Make sure you are requesting to **pull a feature/bugfix branch** (right side). Don't request from master!
- [ ] Check the commit's or even all commits' message styles matches our requested structure.
- [ ] Check your code additions will fail neither code linting checks nor unit test.

### Description
Please describe your pull request.
Screenshot if appropriate to improve evidence.

### Checklist
* [ ] Have you done your changes on a separate branch. Branches MUST have descriptive names that start with either the `fix/` or `feature/` prefixes. Good examples are: `fix/signin-loop` or `feature/pr-templates`.
* [ ] Have you got descriptive commit messages that would make sense if prefixed with "This commit will ..."
* [ ] Will you squash when you merge this PR?

#### New Features

1. [ ] Does your submission have a high coverage of your feature?
2. [ ] Have you lint your code locally prior to submission?

#### Bugs

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests which cover this bug?
* [ ] Have you successfully ran tests with your changes locally?

💔Thank you!

### CODE REVIEW

File: app/src/components/RegisterArtifact.tsx [hunk 1]
```diff
@@ -96,11 +97,18 @@ const RegisterArtifact: React.FC = () => {
...
+    // const metaUrl = IPFS_URL_START + hash;
+
+    const hash = await saveDocumentToArweave(jsonDataBuffer.toString());
+    const metaUrl = ARWEAVE_URL_START + hash;
+
     if (currentAccount === artistAddr) {
       await ArtifactRegistry.methods.mint(
         artistAddr,
-        [artistAddr, IPFS_URL_START + hash],
+        [metaUrl],
```
Comment (DJRHails) [inline app/src/components/RegisterArtifact.tsx]: Why do we remove artistAddr here?

File: app/src/helper/arweave.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+
+const logTransactionData = async (id: string) => {
+  arweave.transactions.getData(id, {decode: true, string: true}).then(data => {
+    console.log(data);
+  });
+};
+
+export const saveDocumentToArweave = async (file: string): Promise<string> => {
+  // TODO: get user's key somehow
```
Comment (DJRHails) [inline app/src/helper/arweave.ts]: Yeah. That was my concern - can we add a hardcoded one yet?

File: app/src/components/RegisterArtifact.tsx (hunk 1 above)
Comment (matt-malarkey) [reply app/src/components/RegisterArtifact.tsx]: oops I think that should still be there

File: app/src/helper/arweave.ts (hunk 2 above)
Comment (matt-malarkey) [reply app/src/helper/arweave.ts]: You can request a wallet that comes with 1AR which is what I've done so far, not sure if you can link between eth wallets or any other useful stuff or whether it's literally just the key for an Arweave specific thing. Are there any support people from Arweave we can talk to through FoB?

Comment (DJRHails) [reply app/src/helper/arweave.ts]: There are - I'll send a message about this through to them

**Pull request merged**